src/cmt_decode_prometheus_remote_write.c

```diff
@@ -0,0 +1,681 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  CMetrics
+ *  ========
+ *  Copyright 2021-2024 The CMetrics Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#include <cmetrics/cmetrics.h>
+#include <cmetrics/cmt_metric.h>
+#include <cmetrics/cmt_map.h>
+#include <cmetrics/cmt_counter.h>
+#include <cmetrics/cmt_summary.h>
+#include <cmetrics/cmt_histogram.h>
+#include <cmetrics/cmt_gauge.h>
+#include <cmetrics/cmt_untyped.h>
+#include <cmetrics/cmt_decode_prometheus_remote_write.h>
+
+static void *__cmt_allocator_alloc(void *data, size_t size) {
+    return malloc(size);
+}
+static void __cmt_allocator_free(void *data, void *ptr) {
+    free(ptr);
+}
+
+static ProtobufCAllocator __cmt_allocator = {
+  .alloc = __cmt_allocator_alloc,
+  .free = __cmt_allocator_free,
+  .allocator_data = NULL
+};
+
+#define cmt_system_allocator __cmt_allocator
+
+static char *cmt_metric_name_from_labels(Prometheus__TimeSeries *ts)
+{
+    int i;
+    int count;
+
+    count = ts->n_labels;
+    for (i = 0; i < count; i++) {
+        if (strncmp("__name__", ts->labels[i]->name, 8) == 0) {
+            return strdup(ts->labels[i]->value);
+        }
+    }
+
+    return NULL;
+}
+
+static struct cmt_map_label *create_map_label(char *caption, size_t length)
+{
+    struct cmt_map_label *map_label;
+
+    map_label = calloc(1, sizeof(struct cmt_map_label));
```

Comment:
I suggest validate the exceptions with returns right away, so it get's easier to read, g.:

```c
    map_label = calloc(1, sizeof(struct cmt_map_label));
    if (!map_label) {
        return NULL;
    }
```    

on this way also we avoid to have another indented block.

Reply:
Fixed in https://github.com/fluent/cmetrics/pull/201/commits/8d6880d91702bd89b46804e1807a323d7d90102e